cmake/Modules/ObsHelpers.cmake

```diff
@@ -546,3 +546,28 @@ function(_install_obs_datatarget target destination)
     COMMENT "Installing ${target} to OBS rundir"
     VERBATIM)
 endfunction()
+
+# Helper function to get all added build targets
```

Comment:
That's not necessary - the `obs` target has all sub-targets added as dependencies (otherwise you couldn't just build that target and get a full OBS installation), so you just need to get the targets recursively from that one, without the need for file-system shenanigans.

So in essence you'd just need to add the `obs` target for a full check, alas we need to filter out the 3rd party ones.

PVS has this already implemented as `pvs_studio_get_recursive_targets`. Only requirement is that you'd need to set up PVS _after_ the OBS target has been fully created.

Reply:
This may be my ignorance in how CMake processes things, but when using `pvs_studio_get_recursive_targets` instead (at the very end of the main `CMakeLists.txt`) I only get the following (on Windows): `OOBS::libobs;Jansson::Jansson;OBS::caption;OBS::w32-pthreads;OBS::frontend-api;OBS::blake2` so plugins themselves appear to be not found that way.

Reply:
Okay so by adding the following to `pvs_studio_get_recursive_targets`:
```diff
    get_target_property(deps "${TARGET}" MANUALLY_ADDED_DEPENDENCIES)
    list(APPEND libs ${deps})
```
I can get *almost* there. The following targets are still missing (when compared to the current method):
```
file-updater
libobs-winrt
media-playback
obs-amf-test
obs-virtualcam-module
opts-parser
```

Reply:
Wouldn't that require us to modify their CMake script whenever they update it, possibly breaking our changes?

Also some targets are header-only libraries, whose source code will be added to a linked target, so there's nothing to "compile" for these targets (and hence check).

Reply:
Right now the script is already slightly modified since by default it would try to add a custom command multiple times for shared dependencies, which causes a fatal cmake error. So that is already problem (perhaps Tracking the changes in a .patch file would be a good idea).

I'm not too worried about header only targets, since it should just don't do anything for those, but the ones currently missing aren't header-only and I'm not entirely sure why are they being omitted.